### PR TITLE
Add IP in IP support

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: calico_containers/adapter/datastore.py [hunk 1]
```diff
@@ -365,12 +405,14 @@ def add_ip_pool(self, version, pool):
...
-        if pool in self.get_ip_pools(version):
-            return
+        key = IP_POOL_KEY % {"version": version,
+                             "pool": str(pool).replace("/", "-")}
 
-        pool_path = IP_POOL_PATH % {"version": version}
-        self.etcd_client.write(pool_path, str(pool), append=True)
+        data = {"cidr" : str(pool)}
+        if ipip:
+            data["ipip"] = "tunl0"
```
Comment (tomdee) [inline calico_containers/adapter/datastore.py]: Do we really want to use a hardcoded value here? tunl0 might conflict with existing tunnels on the host, could we at least make it something a little more calico specific?

Comment (robbrockbank) [inline calico_containers/adapter/datastore.py]: currently mandated by Felix - apparently its the default and is created automatically.  Shaun knows more about this.

File: node_filesystem/templates/felix.cfg.template [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [global]
 MetadataAddr  = None
+IpInIpEnabled = True
```
Comment (tomdee) [inline node_filesystem/templates/felix.cfg.template]: Does this change mean that for calico-docker, we will always get an tunl0 device created? I don't think that's what we want...

Comment (robbrockbank) [inline node_filesystem/templates/felix.cfg.template]: Agree, but leaving as is for MVP.  Will track this getting done at a later date.

**Pull request merged**